src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -358,25 +374,18 @@ public String currentIndexName() {
             return null;
         }
 
-        if (database.updatedAt == null) {
-            return null;
-        }
-
-        return indexNameFor(database.updatedAt.toEpochMilli());
+        return currentIndex;
     }
 
     /**
-     * Index name for a given manifest
+     * New index name for a datasource with current time as suffix
+     *
+     * This method is not thread safe. It could generate same index depends on what Instant.now() returns.
```

Comment:
Any potential issue here for not thread safe? Could we use random str instead of current time to mitigate the thread risk

Reply:
No issue as of current usage. I think I need to remove that comment. It is thread safe, just it does not guarantee the uniqueness of index name.